modules/ingest-common/src/test/java/org/elasticsearch/ingest/common/DateProcessorTests.java

```diff
@@ -105,14 +106,28 @@ public void testJodaPatternLocale() {
     }
 
     public void testJodaPatternDefaultYear() {
+        // fix joda's new DateTime() to Jan 12, 2017
+        DateTimeUtils.setCurrentMillisFixed(1484255848828L);
+
         DateProcessor dateProcessor = new DateProcessor(randomAsciiOfLength(10), DateTimeZone.forID("Europe/Amsterdam"), Locale.ENGLISH,
                 "date_as_string", Collections.singletonList("dd/MM"), "date_as_date");
         Map<String, Object> document = new HashMap<>();
         document.put("date_as_string", "12/06");
+
         IngestDocument ingestDocument = RandomDocumentPicks.randomIngestDocument(random(), document);
         dateProcessor.execute(ingestDocument);
-        assertThat(ingestDocument.getFieldValue("date_as_date", String.class), equalTo(DateTime.now().getYear() +
-                "-06-12T00:00:00.000+02:00"));
+        assertThat(ingestDocument.getFieldValue("date_as_date", String.class), equalTo("2017-06-12T00:00:00.000+02:00"));
+
+        // bump year to check next execution's default year
+        // fix joda's new DateTime() to Jan 12, 2018
+        DateTimeUtils.setCurrentMillisFixed(1515791848828L);
+
+        ingestDocument = RandomDocumentPicks.randomIngestDocument(random(), document);
+        dateProcessor.execute(ingestDocument);
+        assertThat(ingestDocument.getFieldValue("date_as_date", String.class), equalTo("2018-06-12T00:00:00.000+02:00"));
+
+        // cleanup
+        DateTimeUtils.setCurrentMillisSystem();
```

Comment:
sounds like a plan. I thought about it but realized the test is already somewhat making assumptions that things will go right, but I'll wrap it for extra measure 👍